File: examples/shared-element-samples/simple-stack-example-sharedelement-fragment/src/main/java/com/example/fragmenttransitions/BaseFragment.java
package com.example.fragmenttransitions;

import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import android.view.View;

/**
 * Created by Owner on 2017. 08. 08..
 */

public class BaseFragment
        extends Fragment {
    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        startPostponedEnterTransition();
    }

    @NonNull
    public final <T extends BaseKey> T getKey() {
        T key = getArguments() != null ? getArguments().getParcelable("KEY") : null;
        if(key == null) {
            throw new NullPointerException("Key should not be null");
        }
        return key;
    }
}

File: samples/scoping-samples/simple-stack-example-scoping/src/main/java/com/zhuinden/simplestackexamplescoping/core/navigation/BaseFragment.kt
package com.zhuinden.simplestackexamplescoping.core.navigation

import androidx.fragment.app.Fragment
import com.zhuinden.simplestackexamplescoping.utils.requireArguments

/**
 * Created by Zhuinden on 2018.09.17.
 */

open class BaseFragment : Fragment() {
    fun <T : BaseKey> getKey(): T = requireArguments.getParcelable<T>("KEY")
}
